app.py

```diff
@@ -17,15 +23,29 @@ def index():
 @app.route("/books")
 def book_list():
     """Show the list of books"""
-    return render_template("books.html", books=books)
+
+    book_info = {}
+    for id, title, summary, url in db_session.query(Book.id, Book.title, Book.summary, Book.image_url):
```

Comment:
You don't even have to transform the book query result into a dictionary. You can just call `.all()` at the end and pass it into the `books` parameter on line 33. 

See how SQLAlchemy does this in the tutorial, where they call `instance.name` and `instance.fullname`. You can do the same in your template.

https://docs.sqlalchemy.org/en/14/orm/tutorial.html#querying